### PR TITLE
SI-7271 fixes positions of string interpolation parts

### PR DESCRIPTION
**Pull request by xeno-by**
Positions of static parts are now set explicitly during parsing rather
than filled in wholesale during subsequent atPos after parsing.

I also had to change the offsets that scanner uses for initial static
parts of string interpolations so that they no longer point to the
opening double quote, but rather to the actual beginning of the part.

### CODE REVIEW

**Comment by xeno-by** [comment]
review @paulp

File: src/compiler/scala/tools/nsc/ast/parser/Parsers.scala [hunk 1]
```diff
@@ -1152,7 +1152,7 @@ self =>
       val exprBuf = new ListBuffer[Tree]
       in.nextToken()
       while (in.token == STRINGPART) {
-        partsBuf += literal()
+        partsBuf += atPos(in.offset)(literal())
         exprBuf += {
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/ast/parser/Parsers.scala:1156]: Any idea why we wouldn't just do this at literal()? It has the comment "@note  The returned tree does not yet have a position" but fails to give any hint as to why.

Comment (xeno-by) [inline src/compiler/scala/tools/nsc/ast/parser/Parsers.scala:1156]: Good point. Will investigate.

**Comment by xeno-by** [comment]
I thought this is going to be an easy one, but apparently it requires some thought, which I don't have time for at the moment. Therefore I'm closing this PR until better times come.

**Pull request closed without merging**